internal/logging/logger.go

```diff
@@ -69,6 +70,35 @@ func UseFileLogger(filepath string) {
 	L = newLogger(writer)
 }
 
+type logWriter struct {
+	logger interface {
+		WithLevel(level zerolog.Level) *zerolog.Event
+	}
+	level zerolog.Level
+}
+
+func (w logWriter) Write(p []byte) (n int, err error) {
+	n = len(p)
+	p = trimTrailingNewline(p)
+	w.logger.WithLevel(w.level).CallerSkipFrame(1).Msg(string(p))
+	return n, nil
+}
+
+func trimTrailingNewline(p []byte) []byte {
+	n := len(p)
+	if n > 0 && p[n-1] == '\n' {
+		// Trim CR added by stdlog.
+		p = p[0 : n-1]
+	}
+	return p
+}
```

Comment:
This is copied from `zerolog.Logger.Write`. We need to same cleanup applied here because we can't use `Logger.Write`. It emits logs with a nil level.